smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpProtocolGeneratorUtils.java

```diff
@@ -238,4 +242,35 @@ static void generateCollectBodyString(GenerationContext context) {
 
         return errorShapes;
     }
+
+    /**
+     * Writes resolved hostname, prepending existing hostname with hostPrefix and replacing each hostLabel with
+     * the corresponding top-level input member value.
+     *
+     * @param context The generation context.
+     * @param operation The operation to generate for.
+     */
+    static void writeHostPrefix(GenerationContext context, OperationShape operation) {
+        TypeScriptWriter writer = context.getWriter();
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+        EndpointTrait trait = operation.getTrait(EndpointTrait.class).get();
+        writer.write("let resolvedHostname = (context.endpoint as any).hostname;");
+        // Check if disableHostPrefixInjection has been set to true at runtime
+        writer.openBlock("if (context.disableHostPrefix !== true) {", "}", () -> {
+            writer.addImport("validHostname", "__validHostname", "@aws-sdk/util-valid-hostname");
```

Comment:
This could use `AWS_VALID_HOSTNAME.packageName` instead of repeating `"@aws-sdk/util-valid-hostname"`

Reply:
Updated.  Importing from TypeScriptDependency.AWS_SDK_PROTOCOL_HTTP.packageName now.